src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,155 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        // Queues to process skipped events
+        m_deltaQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
```

Comment:
consider using a slice as a template argument.

Reply:
Scenario1: The OS information can not be retrieved and the events are discarded
![image (1)](https://github.com/wazuh/wazuh/assets/13010397/c2c2ff42-b4f9-42d1-95ce-7969483a1962)
Scenario2: The OS information is retrieved and the events are retried. 
![image (2)](https://github.com/wazuh/wazuh/assets/13010397/033dce73-bbcb-48c2-bb2f-7bcebab1094d)